sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -988,4 +1023,12 @@ private int completeMessages(ServiceBusReceiverAsyncClient client, List<String>
 
         return lockTokens.size();
     }
+
+    private void completeDeferredMessages(ServiceBusReceiverAsyncClient client, ServiceBusReceivedMessage receivedMessage) {
```

Comment:
Doesn't this overlap with your pendingDeferredMessages member variable at the top?

Reply:
It does not conflict, because in case of session the receiver has to be the same one which locked the session in first place.